### ISSUE TITLE
Open default terminal from Docker Desktop 

### ISSUE THREAD

**Issue by nebuk89** [docker/roadmap#98]
**Tell us about your request**
When clicking the Cli button the default terminal should open the shell session (iTerm2 in my case)

**Which service(s) is this request for?**
Docker Desktop 

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Today customers have preferred terminals to work in which are setup to work for them, the desktop UI today opens the built in terminal for the machine not the users current default terminal. 

**Are you currently working around the issue?**
Using iTerm rather than button in Desktop UI

**Additional context**
For Mac issue: https://github.com/docker/for-mac/issues/4273

**Comment by brandon-fryslie** [comment]
The button to open a terminal is a brilliant feature, but many users cannot use it due to it opening the wrong terminal application.  I support users who use Docker for Mac, and I would be hesitant to suggest they use while I'm helping them because of the confusion it might cause in opening the wrong terminal application.  Thanks!

**Comment by oocx** [comment]
This request also applies to Windows. I have the new Windows Terminal, conemu cmder,, terminus and hyper installed, but the link on the dashboard opens the outdated plain cmd.exe terminal.

**Comment by macfreek** [comment]
In Docker Toolbox, this used to be possible by changing the `~/Library/Application Support/DockerToolbox/default_terminal` file. Sadly, Docker Desktop is lacking this feature :(.

**Comment by CholtonATX** [comment]
Coming from a place of complete ignorance, this should be perfectly easy to implement. 🤣 

In all sincerity, I would like to be able to pick a shell implementation for myself please. 👍

**Comment by Jeffrey-Meesters** [comment]
TL;DR: +1

Currently it's running this command when using the UI:
`docker exec -it 05048511b4e77d42fda1979fb94b827f9b749bc9b8b0e01eeda284be57721896 /bin/sh; exit`
And opens 2 default bash terminals one with the above command and one that does nothing.

So it even exits right away which means the feature is not usable at all anyway, but would be really nice if it would open up in Iterm & zsh... the users' preffered terminal.

**Comment by strangerting** [comment]
+1

**Comment by Ibsardar** [comment]
+1

Probably you guys have already figured it out, [but just in case I can be of help](https://github.com/docker/for-mac/issues/4273#issuecomment-774755298).

**Comment by stephen-turner** [comment]
This is implemented for iTerm2 in Docker Desktop 3.2.0 and above.

**Comment by stephen-turner** [comment]
Possibly: we did it for iTerm2 first because that was the one we got by far the most requests for.

**Comment by u-foka** [comment]
Hi! Unfortunately it only seems to work if either iTerm isn't running or I have at least one window open. When it is running without any windows open, then it becomes the active application but doesn't open anything.

**Comment by stephen-turner** [comment]
@u-foka Thanks, we'll check.

**Comment by Kazmirchuk** [comment]
+1 for Windows! please add a setting, where we can specify our own command line for opening a new CLI

**Comment by stephen-turner** [comment]
@u-foka Your issue is fixed in version 3.2.2, which we have just released.

**Comment by u-foka** [comment]
@stephen-turner Thanks! Works beautifully now :)

**Comment by sylfabre** [comment]
+1 for Windows!

**Comment by nathan-cummings-ukaea** [comment]
Could this work for macOS users using a different terminal (for example, Alacritty)?

**Comment by nebuk89** [comment]
Windows issue is now open here: https://github.com/docker/roadmap/issues/212 

Please add reactions there for Windows or open something new if we missed something on Mac :D

**Comment by henry-js** [comment]
Windows too please!

**Comment by dmelinosky** [comment]
This would be great! +1

**Comment by hichemfantar** [comment]
Windows terminal support please!

**Comment by tomtranc** [comment]
+1 for Windows terminal support plz.  Is it possible to externalize this with an environment variable so user can set whatever terminal they want to use whenever this CLI button is clicked?

**Comment by thaJeztah** [comment]
Windows is tracked in this issue: https://github.com/docker/roadmap/issues/212

I'll lock the conversation on this issue; subscribe to the other issue if you're arriving here for Windows
